starfish/image/_filter/zero_by_channel_magnitude.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+from copy import deepcopy
+from typing import Optional
+
+import numpy as np
+from tqdm import tqdm
+
+from starfish.constants import Indices, Coordinates
+from starfish.stack import ImageStack
+from ._base import FilterAlgorithmBase
+
+
+class ZeroByChannelMagnitude(FilterAlgorithmBase):
+    def __init__(self, thresh: int, normalize: bool, **kwargs) -> None:
+        """For assays in which we expect codewords to have explicit zero values,
+        e.g., DARTFISH, SEQFISH, etc., this filter allows for the explicit zeroing
+        out of pixels, for each round, where there is insufficient signal magnitude across channels.
+
+
+        Parameters
+        ----------
+        thresh : int
+            pixels in each round that have a L2 norm across channels below this threshold
+            are set to 0
+
+        normalize : bool
+            if True, this scales all rounds to have unit L2 norm across channels
+        """
+        self.thresh = thresh
+        self.normalize = normalize
+
+    @classmethod
+    def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
+        group_parser.add_argument(
+            '--thresh', type=int, help='minimum magnitude threshold for pixels across channels')
+
+    def run(self, stack: ImageStack, in_place: bool = True, verbose=False) -> Optional[ImageStack]:
+        """Perform filtering of an image stack
+
+        Parameters
+        ----------
+        stack : ImageStack
+            Stack to be filtered.
+        in_place : bool
+            if True, process ImageStack in-place, otherwise return a new stack
+        verbose : bool
+            if True, report on the percentage completed during processing (default = False)
+
+        Returns
+        -------
+        Optional[ImageStack] :
+            if in-place is False, return the results of filter as a new stack
+
+        """
+
+        channels_per_round = stack._data.groupby(Indices.ROUND.value)
```

Comment:
@ambrosejcarr note that this is the first filter that does not use apply, since i'm using xarrays groupby feature instead
@ttung note that i'm accessing and manipulating a protected variable, ImageStack._data.

Reply:
@ttung ^^^ still curious what you think about this. i think this will be a common use case